### PR TITLE
Boost instructions

### LINKED ISSUE #32: Replicating/executing build.sh in VSCode
I've been following the [IDE setup guide](https://revbayes.github.io/developer/setup/) to work with RevBayes in VSCode.

One thing it's taken me a while to figure out is that because VSCode doesn't incorporate `build.sh` into its build routine, certain actions need to be performed manually when certain files are changed.

The key aspects are:

- `bash generate_version_number.sh` - generates the version number file
- Move the version number file to the appropriate revbayes cpp directory
- `bash regenerate.sh` - generates updates CMakeLists in order for new / deleted C++ files to be linked during the build process.

I'm not sure what the most appropriate approach is here – to configure VSCode to run a (modified?) version of build.sh, or to manually perform these steps before build.  It would be useful to incorporate guidelines in the [IDE setup guide](https://revbayes.github.io/developer/setup/), if possible.

### PR DESCRIPTION
**Pull request by ms609**
As a CMake novice, I encountered some difficulties relating to Boost whilst installing RevBayes from source.

These additions document my travails, which I hope will assist future users.  Perhaps an expert might spot steps that are redundant, or more efficient troubleshooting guidelines!

I've also added a direct link to the [Windows compliation instructions](https://revbayes.github.io/compile-windows), as I didn't find these instructions straight away, and they were helpful when I did.

### CODE REVIEW

**Comment by bredelings** [comment]
Hi Martin,

Thanks for these improvements!  Let me check first with some developers who use windows (I do not) about how they handle these things.

One possibility is that people are downloading the BOOST libraries from the mingw repositories at http://repo.msys2.org/mingw/x86_64/.

-BenRI

**Comment by ms609** [comment]
Great - the mingw repos look like they could be a much simpler route!

File: home/compile_windows.md [hunk 1]
```diff
@@ -49,6 +49,32 @@ You can also [compile with meson](https://github.com/revbayes/revbayes/blob/deve
...
+    If you see errors, you may need to use a more recent version of b2 than is bundled with Boost:
+    - Download [b2 5.2.0](https://github.com/bfgroup/b2/releases) or above.
+    - Remove (or rename) the local copy of `b2.exe`, so the system uses the updated version installed globally above.
+    - Run `b2 --version` to confirm that a version of b2 >= 5.2 is available.
+    
+    6. Update `<path/to/revbayes>/src/CMakeLists.txt` to point to Boost:
+      a. After the line `project(RevBayes)`, add:
+         `set(BOOST_ROOT "C:/boost/boost_1_82_0/gcc")` 
+         (to match the value specified to b2 in the `prefix` argument)
+      b. After the  line starting `find_package(Boost`, around line 170,
```
Comment (ms609) [inline home/compile_windows.md:76]: I believe this is addressed by 
https://github.com/revbayes/revbayes/pull/498

**Comment by bjoelle** [comment]
@bredelings is this fine for merging or do you want to do some more checks ?

**Review by bredelings — APPROVED** [review]

**Comment by bredelings** [comment]
I was initially concerned that we were describing how to install BOOST in multiple places.  But the instructions are probably a bit different for windows.

Let's merge this now.  We can tweak it later if necessary.

Thanks @ms609 !

**Pull request merged**